### PR TITLE
ELPP-2054 Changes for GOAWS in dev/ci

### PR DESCRIPTION
**Pull request by stephenwf**
@giorgiosironi @elrossco22

### CODE REVIEW

File: salt/pillar/search.sls [hunk 1]
```diff
@@ -1,7 +1,7 @@
 search:
     aws:
-        access_key_id: null
-        secret_access_key: null
+        access_key_id: AKIAIOSFODNN7EXAMPLE
+        secret_access_key: wJalrXUtnFEMI/K7MDENG/bPxRfiCYEXAMPLEKEY
```
Comment (lsh-0) [inline salt/pillar/search.sls]: whoa man, are these dummies?

Comment (stephenwf) [inline salt/pillar/search.sls]: Let me do a quick build and check that it works with the null values with goaws

**Comment by stephenwf** [comment]
@lsh-0 Just confirmed that it works with null values

**Comment by lsh-0** [comment]
ah, cheers. you didn't have to, I should have looked more closely.

File: salt/search/aws.sls [hunk 2]
```diff
@@ -0,0 +1,10 @@
+aws-credentials-cli:
```
Comment (giorgiosironi) [inline salt/search/aws.sls]: the name of this state is misleading as these credentials are used primarily by the PHP SDK (and apparently also by the aws cli tool)

File: salt/search/init.sls [hunk 3]
```diff
@@ -23,6 +23,17 @@ search-repository:
         - require:
             - builder: search-repository
 
+{% if pillar.elife.env in ['dev', 'ci'] %}
+search-queue-create:
+    cmd.run:
+        - name: aws sqs create-queue --region=us-east-1 --queue-name=search--{{ pillar.elife.env }} --endpoint=http://localhost:4100
+        - cwd: /home/{{ pillar.elife.deploy_user.username }}
```
Comment (giorgiosironi) [inline salt/search/init.sls]: is `cwd` necessary?

File: salt/search/aws.sls (hunk 2 above)
Comment (stephenwf) [inline salt/search/aws.sls]: Would just `aws-credentials` suit better?

File: salt/search/init.sls (hunk 3 above)
Comment (stephenwf) [inline salt/search/init.sls]: yes unfortunately due to the `.aws` folder

Comment (giorgiosironi) [inline salt/search/init.sls]: I thought that was found with the `$HOME` environment variable, not with the current working directory?

**Comment by giorgiosironi** [comment]
Merging, testing in `dev`, then enabling in `ci` (that update is not visible here).

**Comment by giorgiosironi** [comment]
Enabled in `ci`, getting some connection refused, investigating.

**Comment by giorgiosironi** [comment]
Was a `mkdir` without permissions, fixing it along with some ordering of states and retrying.

**Comment by giorgiosironi** [comment]
Now green.

**Pull request merged**